cmd/kcp-catalog/bind/catalogentry/bind.go

```diff
@@ -0,0 +1,212 @@
+func (b *BindOptions) Run(ctx context.Context) error {
+	log := klog.FromContext(ctx)
+	config, err := b.ClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	_, currentClusterName, err := pluginhelpers.ParseClusterURL(config.Host)
+	if err != nil {
+		return err
+	}
+
+	// get the base config, which is needed for creation of clients.
+	baseConfig, err := ctrlcfg.GetConfigWithContext("base")
+	if err != nil {
+		return fmt.Errorf("unable to get base config %v", err)
+	}
+
+	// get the entry referenced in the command to which the user wants to bind.
+	path, entryName := logicalcluster.New(b.CatalogEntryRef).Split()
+	client, err := newCatalogClient(baseConfig, path)
+	if err != nil {
+		return err
+	}
+
+	entry := catalogv1alpha1.CatalogEntry{}
+	err = client.Get(context.TODO(), types.NamespacedName{Name: entryName}, &entry)
+	if err != nil {
+		return fmt.Errorf("cannot find the catalog entry %q referenced in the command in the workspace %q", entryName, path)
+	}
+
+	kcpclient, err := newKCPClusterClient(baseConfig, currentClusterName)
+	if err != nil {
+		return err
+	}
+
+	apiBindings := []apisv1alpha1.APIBinding{}
+	for _, ref := range entry.Spec.Exports {
+		// check if ref is valid. Skip if invalid by logging error.
+		if ref.Workspace.Path == "" || ref.Workspace.ExportName == "" {
+			log.Info("invalid reference %q/%q", ref.Workspace.Path, ref.Workspace.ExportName)
+			continue
+		}
+
+		apiBinding := &apisv1alpha1.APIBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: ref.Workspace.ExportName,
+			},
+			Spec: apisv1alpha1.APIBindingSpec{
+				Reference: ref,
+			},
+		}
+
+		apiBindings = append(apiBindings, *apiBinding)
+	}
+
+	// Create bindings to the target workspace
+	for _, binding := range apiBindings {
+		err := kcpclient.Create(ctx, &binding)
+		if err != nil {
+			// If an APIBinding already exists, intentionally not updating it since we would not like reset AcceptablePermissionClaims.
+			klog.Infof("Failed to create API binding %s: %w", binding.Name, err)
+			continue
+		}
+	}
+
+	if err := wait.PollImmediate(time.Millisecond*500, b.BindWaitTimeout, func() (done bool, err error) {
+		availableBindings := []apisv1alpha1.APIBinding{}
+		for _, binding := range apiBindings {
+			createdBinding := apisv1alpha1.APIBinding{}
+			err = kcpclient.Get(ctx, types.NamespacedName{Name: binding.Name}, &createdBinding)
+			if err != nil {
+				return false, err
+			}
+			availableBindings = append(availableBindings, createdBinding)
+		}
+		return bindReady(availableBindings), nil
+	}); err != nil {
+		return fmt.Errorf("bindings for catalog entry %s could not be created successfully %v", entryName, err)
+	}
+
+	if _, err := fmt.Fprintf(b.Out, "%s created and bound to catalog entry.\n", entryName); err != nil {
+		return err
+	}
+	return nil
+}
+
+func bindReady(bindings []apisv1alpha1.APIBinding) bool {
+	for _, binding := range bindings {
+		if binding.Status.Phase != apisv1alpha1.APIBindingPhaseBound {
```

Comment:
(no action required) I would probably check for the InitialBindingCompleted or BindingUpToDate condition, but this works too